link/src/main/java/com/stripe/android/link/ui/signup/SignUpViewModel.kt

```diff
@@ -45,9 +45,12 @@ internal class SignUpViewModel @Inject constructor(
     /**
      * Emits the email entered in the form if valid, null otherwise.
      */
-    private val consumerEmail: StateFlow<String?> = emailElement.getFormFieldValueFlow().map {
-        it.firstOrNull()?.second?.takeIf { formFieldEntry -> formFieldEntry.isComplete }?.value
-    }.stateIn(viewModelScope, SharingStarted.Lazily, args.customerEmail)
+    private val consumerEmail: StateFlow<String?> =
+        emailElement.getFormFieldValueFlow().map { formFieldsList ->
+            // formFieldsList contains only one element, for the email. Take the second value of
```

Comment:
glad you added this comment. 💯 

Is there anything cleaner we can do here?

Reply:
Not really, I think you just need to understand what the element emits.